python/ray/exceptions.py

```diff
@@ -147,13 +148,48 @@ def __init__(
 
         assert traceback_str is not None
 
-    def make_dual_exception_type(self) -> Type:
-        """Makes a Type that inherits from both RayTaskError and the type of
+    def make_dual_exception_instance(self) -> "RayTaskError":
+        """Makes a object instance that inherits from both RayTaskError and the type of
```

Comment:
Maybe add a line of why we distinguish out ExceptionGroup. Something like:

For normal user `Exception`s, we subclass from both RayTaskError and the user exception. For `ExceptionGroup`, we special handle it because it has a different __new__() signature [1] from `Exception`.

[1] https://docs.python.org/3/reference/datamodel.html#object.__new__

Reply:
done